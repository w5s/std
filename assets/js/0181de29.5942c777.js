"use strict";(self.webpackChunk_w5s_website=self.webpackChunk_w5s_website||[]).push([[4433],{1184:(e,n,s)=>{s.d(n,{R:()=>a,x:()=>c});var t=s(4041);const l={},i=t.createContext(l);function a(e){const n=t.useContext(i);return t.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function c(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(l):e.components||l:a(e.components),t.createElement(i.Provider,{value:n},e.children)}},3110:(e,n,s)=>{s.r(n),s.d(n,{contentTitle:()=>a,default:()=>r,frontMatter:()=>i,toc:()=>c});var t=s(1085),l=s(1184);const i={},a="W5S Iterable and AsyncIterable modules (@w5s/iterable)",c=[{value:"Installation",id:"installation",level:2},{value:"Usage",id:"usage",level:2},{value:"License",id:"license",level:2}];function o(e){const n={a:"a",code:"code",em:"em",h1:"h1",h2:"h2",header:"header",img:"img",p:"p",pre:"pre",...(0,l.R)(),...e.components};return(0,t.jsxs)(t.Fragment,{children:[(0,t.jsx)(n.header,{children:(0,t.jsxs)(n.h1,{id:"w5s-iterable-and-asynciterable-modules-w5siterable",children:["W5S Iterable and AsyncIterable modules ",(0,t.jsx)(n.em,{children:"(@w5s/iterable)"})]})}),"\n",(0,t.jsxs)(n.p,{children:[(0,t.jsx)(n.a,{href:"https://www.npmjs.com/package/@w5s/iterable",children:(0,t.jsx)(n.img,{src:"https://img.shields.io/npm/v/@w5s/iterable.svg?style=flat-square",alt:"NPM Version"})}),"\n",(0,t.jsx)(n.a,{href:"https://www.npmjs.com/package/@w5s/iterable",children:(0,t.jsx)(n.img,{src:"https://img.shields.io/badge/license-MIT-green.svg?style=flat-square",alt:"License"})})]}),"\n",(0,t.jsx)(n.h2,{id:"installation",children:"Installation"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-sh",children:"npm install @w5s/iterable\n"})}),"\n",(0,t.jsx)(n.h2,{id:"usage",children:"Usage"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-ts",children:"import { objectId } from '@w5s/object-id';\n\nexport function main(): void {\n  const someObject = { foo: 1 };\n  const id = objectId(someObject); // typeof id === 'number'\n  console.log(id === objectId(someObject)); // true (because idempotent)\n}\n"})}),"\n",(0,t.jsx)(n.h2,{id:"license",children:"License"}),"\n",(0,t.jsxs)(n.p,{children:[(0,t.jsx)(n.a,{href:"https://www.npmjs.com/package/@w5s/iterable",children:"MIT"})," \xa9 Julien Polo ",(0,t.jsx)(n.a,{href:"mailto:julien.polo@gmail.com",children:"julien.polo@gmail.com"})]})]})}function r(e={}){const{wrapper:n}={...(0,l.R)(),...e.components};return n?(0,t.jsx)(n,{...e,children:(0,t.jsx)(o,{...e})}):o(e)}}}]);