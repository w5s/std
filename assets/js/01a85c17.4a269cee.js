"use strict";(self.webpackChunk_w5s_website=self.webpackChunk_w5s_website||[]).push([[8412],{7204:(e,t,s)=>{s.d(t,{c:()=>p});s(5668);var a=s(5200),i=s(4308),r=s(776),l=s(2984),n=s(5976),c=s(6868);const o={sidebar:"sidebar_RYHo",sidebarItemTitle:"sidebarItemTitle_sRjx",sidebarItemList:"sidebarItemList_uMtB",sidebarItem:"sidebarItem_rygH",sidebarItemLink:"sidebarItemLink_EKgd",sidebarItemLinkActive:"sidebarItemLinkActive_hRXJ"};var d=s(7e3);function m(e){let{sidebar:t}=e;const s=(0,c.o)(t.items);return(0,d.jsx)("aside",{className:"col col--3",children:(0,d.jsxs)("nav",{className:(0,a.c)(o.sidebar,"thin-scrollbar"),"aria-label":(0,n.G)({id:"theme.blog.sidebar.navAriaLabel",message:"Blog recent posts navigation",description:"The ARIA label for recent posts in the blog sidebar"}),children:[(0,d.jsx)("div",{className:(0,a.c)(o.sidebarItemTitle,"margin-bottom--md"),children:t.title}),(0,d.jsx)("ul",{className:(0,a.c)(o.sidebarItemList,"clean-list"),children:s.map((e=>(0,d.jsx)("li",{className:o.sidebarItem,children:(0,d.jsx)(l.default,{isNavLink:!0,to:e.permalink,className:o.sidebarItemLink,activeClassName:o.sidebarItemLinkActive,children:e.title})},e.permalink)))})]})})}var u=s(5240);function g(e){let{sidebar:t}=e;const s=(0,c.o)(t.items);return(0,d.jsx)("ul",{className:"menu__list",children:s.map((e=>(0,d.jsx)("li",{className:"menu__list-item",children:(0,d.jsx)(l.default,{isNavLink:!0,to:e.permalink,className:"menu__link",activeClassName:"menu__link--active",children:e.title})},e.permalink)))})}function b(e){return(0,d.jsx)(u.Mx,{component:g,props:e})}function h(e){let{sidebar:t}=e;const s=(0,r.U)();return t?.items.length?"mobile"===s?(0,d.jsx)(b,{sidebar:t}):(0,d.jsx)(m,{sidebar:t}):null}function p(e){const{sidebar:t,toc:s,children:r,...l}=e,n=t&&t.items.length>0;return(0,d.jsx)(i.c,{...l,children:(0,d.jsx)("div",{className:"container margin-vert--lg",children:(0,d.jsxs)("div",{className:"row",children:[(0,d.jsx)(h,{sidebar:t}),(0,d.jsx)("main",{className:(0,a.c)("col",{"col--7":n,"col--9 col--offset-1":!n}),itemScope:!0,itemType:"https://schema.org/Blog",children:r}),s&&(0,d.jsx)("div",{className:"col col--2",children:s})]})})})}},3664:(e,t,s)=>{s.r(t),s.d(t,{default:()=>h});s(5668);var a=s(5200),i=s(7676),r=s(7520),l=s(3692),n=s(7204),c=s(5480),o=s(4836);const d={tag:"tag_YYDp"};var m=s(7e3);function u(e){let{letterEntry:t}=e;return(0,m.jsxs)("article",{children:[(0,m.jsx)(o.default,{as:"h2",id:t.letter,children:t.letter}),(0,m.jsx)("ul",{className:"padding--none",children:t.tags.map((e=>(0,m.jsx)("li",{className:d.tag,children:(0,m.jsx)(c.c,{...e})},e.permalink)))}),(0,m.jsx)("hr",{})]})}function g(e){let{tags:t}=e;const s=(0,i.M)(t);return(0,m.jsx)("section",{className:"margin-vert--lg",children:s.map((e=>(0,m.jsx)(u,{letterEntry:e},e.letter)))})}var b=s(7840);function h(e){let{tags:t,sidebar:s}=e;const c=(0,i.K)();return(0,m.jsxs)(r.cr,{className:(0,a.c)(l.W.wrapper.blogPages,l.W.page.blogTagsListPage),children:[(0,m.jsx)(r.U7,{title:c}),(0,m.jsx)(b.c,{tag:"blog_tags_list"}),(0,m.jsxs)(n.c,{sidebar:s,children:[(0,m.jsx)(o.default,{as:"h1",children:c}),(0,m.jsx)(g,{tags:t})]})]})}},5480:(e,t,s)=>{s.d(t,{c:()=>n});s(5668);var a=s(5200),i=s(2984);const r={tag:"tag_qE9H",tagRegular:"tagRegular_aHXt",tagWithCount:"tagWithCount_UC8q"};var l=s(7e3);function n(e){let{permalink:t,label:s,count:n}=e;return(0,l.jsxs)(i.default,{href:t,className:(0,a.c)(r.tag,n?r.tagWithCount:r.tagRegular),children:[s,n&&(0,l.jsx)("span",{children:n})]})}},6868:(e,t,s)=>{s.d(t,{o:()=>l});var a=s(5668),i=s(7672),r=s(1144);function l(e){const{pathname:t}=(0,i.IT)();return(0,a.useMemo)((()=>e.filter((e=>function(e,t){return!(e.unlisted&&!(0,r.Sc)(e.permalink,t))}(e,t)))),[e,t])}},7676:(e,t,s)=>{s.d(t,{K:()=>i,M:()=>r});var a=s(5976);const i=()=>(0,a.G)({id:"theme.tags.tagsPageTitle",message:"Tags",description:"The title of the tag list page"});function r(e){const t={};return Object.values(e).forEach((e=>{const s=function(e){return e[0].toUpperCase()}(e.label);t[s]??=[],t[s].push(e)})),Object.entries(t).sort(((e,t)=>{let[s]=e,[a]=t;return s.localeCompare(a)})).map((e=>{let[t,s]=e;return{letter:t,tags:s.sort(((e,t)=>e.label.localeCompare(t.label)))}}))}}}]);