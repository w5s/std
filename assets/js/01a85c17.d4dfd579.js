"use strict";(self.webpackChunk_w5s_website=self.webpackChunk_w5s_website||[]).push([[4013],{3485:(e,t,s)=>{s.d(t,{Z:()=>p});s(2784);var a=s(524),i=s(3144),l=s(7963),r=s(9817),n=s(1077),c=s(3852);const d={sidebar:"sidebar_RYHo",sidebarItemTitle:"sidebarItemTitle_sRjx",sidebarItemList:"sidebarItemList_uMtB",sidebarItem:"sidebarItem_rygH",sidebarItemLink:"sidebarItemLink_EKgd",sidebarItemLinkActive:"sidebarItemLinkActive_hRXJ"};var o=s(2322);function m(e){let{sidebar:t}=e;const s=(0,c.c)(t.items);return(0,o.jsx)("aside",{className:"col col--3",children:(0,o.jsxs)("nav",{className:(0,a.Z)(d.sidebar,"thin-scrollbar"),"aria-label":(0,n.I)({id:"theme.blog.sidebar.navAriaLabel",message:"Blog recent posts navigation",description:"The ARIA label for recent posts in the blog sidebar"}),children:[(0,o.jsx)("div",{className:(0,a.Z)(d.sidebarItemTitle,"margin-bottom--md"),children:t.title}),(0,o.jsx)("ul",{className:(0,a.Z)(d.sidebarItemList,"clean-list"),children:s.map((e=>(0,o.jsx)("li",{className:d.sidebarItem,children:(0,o.jsx)(r.default,{isNavLink:!0,to:e.permalink,className:d.sidebarItemLink,activeClassName:d.sidebarItemLinkActive,children:e.title})},e.permalink)))})]})})}var u=s(7548);function g(e){let{sidebar:t}=e;const s=(0,c.c)(t.items);return(0,o.jsx)("ul",{className:"menu__list",children:s.map((e=>(0,o.jsx)("li",{className:"menu__list-item",children:(0,o.jsx)(r.default,{isNavLink:!0,to:e.permalink,className:"menu__link",activeClassName:"menu__link--active",children:e.title})},e.permalink)))})}function b(e){return(0,o.jsx)(u.Zo,{component:g,props:e})}function h(e){let{sidebar:t}=e;const s=(0,l.i)();return t?.items.length?"mobile"===s?(0,o.jsx)(b,{sidebar:t}):(0,o.jsx)(m,{sidebar:t}):null}function p(e){const{sidebar:t,toc:s,children:l,...r}=e,n=t&&t.items.length>0;return(0,o.jsx)(i.Z,{...r,children:(0,o.jsx)("div",{className:"container margin-vert--lg",children:(0,o.jsxs)("div",{className:"row",children:[(0,o.jsx)(h,{sidebar:t}),(0,o.jsx)("main",{className:(0,a.Z)("col",{"col--7":n,"col--9 col--offset-1":!n}),itemScope:!0,itemType:"https://schema.org/Blog",children:l}),s&&(0,o.jsx)("div",{className:"col col--2",children:s})]})})})}},5832:(e,t,s)=>{s.r(t),s.d(t,{default:()=>h});s(2784);var a=s(524),i=s(4103),l=s(8162),r=s(211),n=s(3485),c=s(4178),d=s(3851);const o={tag:"tag_YYDp"};var m=s(2322);function u(e){let{letterEntry:t}=e;return(0,m.jsxs)("article",{children:[(0,m.jsx)(d.default,{as:"h2",id:t.letter,children:t.letter}),(0,m.jsx)("ul",{className:"padding--none",children:t.tags.map((e=>(0,m.jsx)("li",{className:o.tag,children:(0,m.jsx)(c.Z,{...e})},e.permalink)))}),(0,m.jsx)("hr",{})]})}function g(e){let{tags:t}=e;const s=(0,i.P)(t);return(0,m.jsx)("section",{className:"margin-vert--lg",children:s.map((e=>(0,m.jsx)(u,{letterEntry:e},e.letter)))})}var b=s(4390);function h(e){let{tags:t,sidebar:s}=e;const c=(0,i.M)();return(0,m.jsxs)(l.FG,{className:(0,a.Z)(r.k.wrapper.blogPages,r.k.page.blogTagsListPage),children:[(0,m.jsx)(l.d,{title:c}),(0,m.jsx)(b.Z,{tag:"blog_tags_list"}),(0,m.jsxs)(n.Z,{sidebar:s,children:[(0,m.jsx)(d.default,{as:"h1",children:c}),(0,m.jsx)(g,{tags:t})]})]})}},4178:(e,t,s)=>{s.d(t,{Z:()=>n});s(2784);var a=s(524),i=s(9817);const l={tag:"tag_qE9H",tagRegular:"tagRegular_aHXt",tagWithCount:"tagWithCount_UC8q"};var r=s(2322);function n(e){let{permalink:t,label:s,count:n}=e;return(0,r.jsxs)(i.default,{href:t,className:(0,a.Z)(l.tag,n?l.tagWithCount:l.tagRegular),children:[s,n&&(0,r.jsx)("span",{children:n})]})}},3852:(e,t,s)=>{s.d(t,{c:()=>r});var a=s(2784),i=s(7267),l=s(7661);function r(e){const{pathname:t}=(0,i.TH)();return(0,a.useMemo)((()=>e.filter((e=>function(e,t){return!(e.unlisted&&!(0,l.Mg)(e.permalink,t))}(e,t)))),[e,t])}},4103:(e,t,s)=>{s.d(t,{M:()=>i,P:()=>l});var a=s(1077);const i=()=>(0,a.I)({id:"theme.tags.tagsPageTitle",message:"Tags",description:"The title of the tag list page"});function l(e){const t={};return Object.values(e).forEach((e=>{const s=function(e){return e[0].toUpperCase()}(e.label);t[s]??=[],t[s].push(e)})),Object.entries(t).sort(((e,t)=>{let[s]=e,[a]=t;return s.localeCompare(a)})).map((e=>{let[t,s]=e;return{letter:t,tags:s.sort(((e,t)=>e.label.localeCompare(t.label)))}}))}}}]);