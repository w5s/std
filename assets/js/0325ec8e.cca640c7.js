"use strict";(self.webpackChunk_w5s_website=self.webpackChunk_w5s_website||[]).push([[7654],{3914:t=>{t.exports=JSON.parse('{"categoryGeneratedIndex":{"title":"Getting started","description":"5 minutes to learn the most important Docusaurus concepts.","slug":"/category/getting-started","permalink":"/std/docs/category/getting-started","sidebar":"tutorialSidebar","navigation":{"next":{"title":"Introduction","permalink":"/std/docs/getting-started/Introduction"}}}}')}}]);