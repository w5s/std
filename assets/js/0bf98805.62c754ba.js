"use strict";(self.webpackChunk_w5s_website=self.webpackChunk_w5s_website||[]).push([[9635],{3769:e=>{e.exports=JSON.parse('{"name":"docusaurus-plugin-content-docs","id":"default"}')}}]);