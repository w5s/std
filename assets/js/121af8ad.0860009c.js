"use strict";(self.webpackChunk_w5s_website=self.webpackChunk_w5s_website||[]).push([[4561],{4494:e=>{e.exports=JSON.parse('{"categoryGeneratedIndex":{"title":"Error","slug":"/category/error","permalink":"/std/docs/category/error","sidebar":"tutorialSidebar","navigation":{"previous":{"title":"Packages","permalink":"/std/docs/category/packages"},"next":{"title":"Overview","permalink":"/std/docs/packages/error/overview"}}}}')}}]);