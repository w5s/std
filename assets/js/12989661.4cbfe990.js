"use strict";(self.webpackChunk_w5s_website=self.webpackChunk_w5s_website||[]).push([[5939],{7307:(e,n,s)=>{s.r(n),s.d(n,{contentTitle:()=>o,default:()=>c,frontMatter:()=>r,toc:()=>i});var t=s(1085),a=s(1184);const r={},o="W5S Task modules (@w5s/task)",i=[{value:"Installation",id:"installation",level:2},{value:"Usage",id:"usage",level:2},{value:"Example",id:"example",level:3},{value:"License",id:"license",level:2}];function l(e){const n={a:"a",code:"code",em:"em",h1:"h1",h2:"h2",h3:"h3",header:"header",img:"img",p:"p",pre:"pre",...(0,a.R)(),...e.components};return(0,t.jsxs)(t.Fragment,{children:[(0,t.jsx)(n.header,{children:(0,t.jsxs)(n.h1,{id:"w5s-task-modules-w5stask",children:["W5S Task modules ",(0,t.jsx)(n.em,{children:"(@w5s/task)"})]})}),"\n",(0,t.jsxs)(n.p,{children:[(0,t.jsx)(n.a,{href:"https://www.npmjs.com/package/@w5s/task",children:(0,t.jsx)(n.img,{src:"https://img.shields.io/npm/v/@w5s/task.svg?style=flat-square",alt:"NPM Version"})}),"\n",(0,t.jsx)(n.a,{href:"https://github.com/w5s/project-config/blob/HEAD/packages/core/LICENSE",children:(0,t.jsx)(n.img,{src:"https://img.shields.io/badge/license-MIT-green.svg?style=flat-square",alt:"License"})})]}),"\n",(0,t.jsx)(n.h2,{id:"installation",children:"Installation"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-sh",children:"npm install @w5s/task\n"})}),"\n",(0,t.jsx)(n.h2,{id:"usage",children:"Usage"}),"\n",(0,t.jsx)(n.h3,{id:"example",children:"Example"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-typescript",children:"import { Result } from '@w5s/core';\nimport { Task } from '@w5s/task';\n\nfunction parseNumber(expr: string) {\n  const parsed = Number(expr);\n\n  // - Return a immutable Result object\n  // - Avoid throwing error because impure\n  // - Avoid using NaN because the error case is implicit in the typing\n  return Number.isNaN(parsed) ? Result.Ok(parsed) : Result.Error('NotANumber');\n}\n\nfunction log(value: unknown) {\n  // This is a lazy operation that will only be evaluated when the Task is run\n  return Task.create(({ ok }) => ok(console.log(value)));\n}\n\nexport function main() {\n  const parsed = parseNumber('1.1'); // Result.Ok(1.1)\n  const computed = Result.map(parsed, (amount) => amount + 2); // Result.Ok(3.1)\n\n  // Lazy operation that will display in console the computed result when evaluated\n  return log(computed);\n}\n\n// runTask is impure and should be put at the edge of the program\nvoid Task.run(main()); // prints { _: 'Result/Ok', value: 3.1 }\n"})}),"\n",(0,t.jsx)(n.h2,{id:"license",children:"License"}),"\n",(0,t.jsxs)(n.p,{children:[(0,t.jsx)(n.a,{href:"https://github.com/w5s/project-config/blob/HEAD/packages/core/LICENSE",children:"MIT"})," \xa9 Julien Polo ",(0,t.jsx)(n.a,{href:"mailto:julien.polo@gmail.com",children:"julien.polo@gmail.com"})]})]})}function c(e){void 0===e&&(e={});const{wrapper:n}={...(0,a.R)(),...e.components};return n?(0,t.jsx)(n,{...e,children:(0,t.jsx)(l,{...e})}):l(e)}},1184:(e,n,s)=>{s.d(n,{R:()=>o,x:()=>i});var t=s(4041);const a={},r=t.createContext(a);function o(e){const n=t.useContext(r);return t.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function i(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(a):e.components||a:o(e.components),t.createElement(r.Provider,{value:n},e.children)}}}]);