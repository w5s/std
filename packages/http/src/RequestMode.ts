import type { Enum } from '@w5s/core';
import { define } from '@w5s/core/dist/Enum/define.js';

export const RequestMode = define({
  typeName: 'RequestMode',
  SameOrigin: 'same-origin',
  CORS: 'cors',
  Navigate: 'navigate',
  NoCORS: 'no-cors',
});
/**
 * HTTP referrer policy type
 *
 * @see https://developer.mozilla.org/en-US/docs/Web/API/Request/referrerPolicy
 */
export type RequestMode = Enum.ValueOf<typeof RequestMode>;
