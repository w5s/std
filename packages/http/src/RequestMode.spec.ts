import { describe, expect, it } from 'vitest';
import { RequestMode } from './RequestMode.js';

describe('RequestMode', () => {
  it('is an enum', () => {
    expect(RequestMode).toMatchSnapshot();
  });
});
