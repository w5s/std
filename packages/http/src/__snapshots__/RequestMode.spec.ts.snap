// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RequestMode > is an enum 1`] = `
{
  "CORS": "cors",
  "Navigate": "navigate",
  "NoCORS": "no-cors",
  "SameOrigin": "same-origin",
  "at": [Function],
  "codecDecode": [Function],
  "codecEncode": [Function],
  "codecSchema": [Function],
  "from": [Function],
  "hasInstance": [Function],
  "indexOf": [Function],
  "indexType": "number",
  "inspect": undefined,
  "range": [Function],
  "rangeSize": [Function],
  "typeName": "RequestMode",
  Symbol(w5s.enumKeys): [
    "SameOrigin",
    "CORS",
    "Navigate",
    "NoCORS",
  ],
}
`;
